go/libraries/doltcore/sqle/tables.go

```diff
@@ -1680,16 +1682,17 @@ func (t *AlterableDoltTable) isIncompatibleTypeChange(oldColumn *sql.Column, new
 	}
 
 	if !existingCol.TypeInfo.Equals(newCol.TypeInfo) {
-		if types.IsFormat_DOLT(t.Format()) {
-			// This is overly broad, we could narrow this down a bit
-			return true
-		}
-		if existingCol.Kind != newCol.Kind {
-			return true
-		} else if schema.IsColSpatialType(newCol) {
+		if schema.IsColSpatialType(newCol) {
 			// TODO: we need to do this because some spatial type changes require a full table check, but not all.
 			//  We could narrow this check down.
 			return true
+		} else if types.IsFormat_DOLT(t.Format()) {
```

Comment:
(minor) not a big deal, but we can get rid of these checks for the "new" storage format and the code below in the else if block checking `existingCol.Kind != newCol.Kind`, now that we've been on new/dolt storage format for a while.